src/js/components/modal.js

```diff
@@ -0,0 +1,265 @@
+const behavior = require("../utils/behavior");
+const select = require("../utils/select");
+const FocusTrap = require("../utils/focus-trap");
+const ScrollBarWidth = require("../utils/scrollbar-width");
+
+const { CLICK } = require("../events");
+const { prefix: PREFIX } = require("../config");
+
+const MODAL_CLASSNAME = `${PREFIX}-modal`;
+const OVERLAY_CLASSNAME = `${MODAL_CLASSNAME}-overlay`;
+const WRAPPER_CLASSNAME = `${MODAL_CLASSNAME}-wrapper`;
+const OPENER_ATTRIBUTE = "data-open-modal";
+const CLOSER_ATTRIBUTE = "data-close-modal";
+const FORCE_ACTION_ATTRIBUTE = "data-force-action";
+const MODAL = `.${MODAL_CLASSNAME}`;
+const INITIAL_FOCUS = `.${WRAPPER_CLASSNAME} *[data-focus]`;
+const CLOSE_BUTTON = `${WRAPPER_CLASSNAME} *[${CLOSER_ATTRIBUTE}]`;
+const OPENERS = `*[${OPENER_ATTRIBUTE}][aria-controls]`;
+const CLOSERS = `${CLOSE_BUTTON}, .${OVERLAY_CLASSNAME}:not([${FORCE_ACTION_ATTRIBUTE}])`;
+
+const ACTIVE_CLASS = "usa-js-mobile-nav--active";
+const PREVENT_CLICK_CLASS = "usa-js-no-click";
+const VISIBLE_CLASS = "is-visible";
+const HIDDEN_CLASS = "is-hidden";
+
+const nonModals = document.querySelectorAll(
+  `body > *:not(${MODAL}):not([aria-hidden])`
+);
+
+let modal;
+
+const isActive = () => document.body.classList.contains(ACTIVE_CLASS);
+const SCROLLBAR_WIDTH = ScrollBarWidth();
+const INITIAL_PADDING = window
+  .getComputedStyle(document.body)
+  .getPropertyValue("padding-right");
+const TEMPORARY_PADDING =
+  parseInt(INITIAL_PADDING.replace(/px/, ""), 10) +
+  parseInt(SCROLLBAR_WIDTH.replace(/px/, ""), 10) +
+  "px";
+
+/**
+ *  Is bound to escape key, closes modal when
+ */
+const onMenuClose = () => {
+  modal.toggleModal.call(modal, false);
+};
+
+/**
+ *  Toggle the visibility of a modal window
+ *
+ * @param {KeyboardEvent} event the keydown event
+ * @returns {boolean} safeActive if mobile is open
+ */
+function toggleModal(event) {
+  let originalOpener;
+  let clickedElement = event.target;
+  const { body } = document;
+  const safeActive = !isActive();
+  const modalId = clickedElement
+    ? clickedElement.getAttribute("aria-controls")
+    : document.querySelector(".usa-modal-wrapper.is-visible");
+  const targetModal = safeActive
+    ? document.getElementById(modalId)
+    : document.querySelector(".usa-modal-wrapper.is-visible");
+  const openFocusEl = targetModal.querySelector(INITIAL_FOCUS)
```

Comment:
Modal stops responding to clicks on the slightest resize.  

https://user-images.githubusercontent.com/3385219/109984362-4ef6cd00-7cc9-11eb-8fe0-a7dddf571aae.mp4

<details>
  <summary>❗ Console error</summary>

```
Uncaught TypeError: can't access property "querySelector", i is null
    L modal.js:66
    exports index.js:8
    exports index.js:4
    exports index.js:3
    simulateClickEffect browser-sync-client.js:10
    setTimeout handler*8558/e.simulateClickEffect/< browser-sync-client.js:10
    __tryOrSetError browser-sync-client.js:10
    next browser-sync-client.js:10
    _next browser-sync-client.js:10
    next browser-sync-client.js:10
modal.js:66:22
```

</details>

---

Reproduced in:
- Firefox 87.0b4 (64-bit)
- Chrome 89.0.4389.72

Reply:
I see what's happening here. The modal uses the same body class and logic (`safeActive`) as the navigation when open: `usa-js-mobile-nav--active`. When you resize the window, the navigation js removes that class, so the modal thinks it should be opening a window instead of closing it, so it errors out.

I think a fix for this might just to create a new selector `usa-js-modal--active` and use that for `ACTIVE_CLASS`. It'd need a CSS selctor, like https://github.com/uswds/uswds/blob/781b9d00abf70eec9930333016b6999d9eca8a47/src/stylesheets/components/_navigation.scss#L349

Reply:
Here's an example of the modal successfully closing when `usa-js-mobile-nav--active` is added it back to the `<body>`:

https://share.getcloudapp.com/2NuEo1Ak

So I think what is described above should do the trick and eliminate the scripts from colliding.

Reply:
Thank you! Added this in a21ce77.